src/mots/directory.py

```diff
@@ -0,0 +1,254 @@
+            logger.debug("People directory modified, updating configuration...")
+            self.config_handle.config["people"] = self.people.serialized
+
+    def query(self, *paths: str) -> QueryResult:
+        """Query given paths and return a list of corresponding modules."""
+        result = {}
+        rejected = []
+        for path in paths:
+            if not (self.repo_path / path).exists():
+                logger.warning(f"Path {path} does not exist, skipping.")
+                rejected.append(path)
+                continue
+            result[path] = self.index.get(self.repo_path / path, list())
+        logger.debug(f"Query {paths} resolved to {result}.")
+
+        return QueryResult(result, rejected)
+
+    def _export_wiki(self):
+        from jinja2 import Template
+
+        template = Template(WIKI_TMPL)
+        out = template.render(directory=self)
+        return out
+
+    def export(self, frmt="wiki"):
+        """Export directory in a specified format."""
+        supported_formats = ["wiki"]
+        if frmt not in supported_formats:
+            raise ValueError(f"{frmt} not one of {supported_formats}.")
+
+        return getattr(self, f"_export_{frmt}")()
+
+
+class QueryResult:
+    """Helper class to simplify query result interpretation."""
+
+    paths: list = None
+    path_map: dict = None
+    rejected_paths: list = None
+    modules: list = None
+    owners: list = None
+    peers: list = None
+
+    data_keys = {
+        "paths",
+        "rejected_paths",
+        "modules",
+        "owners",
+        "peers",
+    }
+
+    def __init__(self, result, rejected):
+        data = {k: list() for k in self.data_keys}
+        self.path_map = result
+
+        for path in self.path_map:
+            data["paths"].append(path)
+            data["modules"] += self.path_map[path]
+
+        for module in data["modules"]:
+            # TODO: this conversion should happen elsewhere.
+            data["owners"] += [Person(**o) for o in module.owners]
+            data["peers"] += [Person(**p) for p in module.peers]
+
+        data["rejected_paths"] = rejected
+
+        # Remove duplicate entries in all data attributes.
+        for key in data:
+            setattr(self, key, list(set(data[key])))
+
+    def __add__(self, query_result):
+        """Merge the data from both QueryResult objects."""
+        path_map = self.path_map.copy()
+        path_map.update(query_result.path_map)
+        rejected_paths = set.intersection(
+            set(self.rejected_paths),
+            set(query_result.rejected_paths),
+        )
+        return QueryResult(path_map, rejected_paths)
+
+    def __radd__(self, query_result):
+        """Call self.__add__ since the order of addition does not matter."""
+        return self.__add__(query_result)
+
+
+@dataclass
+class Person:
+    """A class representing a person."""
+
+    bmo_id: int = None
```

Comment:
If you'd prefer to keep them non-optional (which is good IMO), I'd recommend doing `assert bmo_data` instead of `if bmo_data` in `__post_init__()`, so that if `bmo_data` is `None` then an error will be raised, rather than having an empty `Person` object.